src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -15,4 +34,66 @@ public String greeting(
         model.addAttribute("name", name);
         return "greeting";
     }
+
+    @GetMapping
+    public ResponseEntity<List<Book>> getAllBooks() {
+        List<Book> response = bookRepository.findAll();
+        if (response.isEmpty()) {
+            return ResponseEntity.noContent().build();
+        }
+        return ResponseEntity.ok(response);
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<Book> getBookById(@PathVariable("id") Long id) {
+        Optional<Book> response = bookRepository.findById(id);
+        if(response.isPresent()) {
+            return ResponseEntity.ok(response.get());
+        } else {
+            throw new BookNotFoundException(MSG);
+        }
+    }
+
+    @GetMapping("/byAuthor")
```

Comment:
You can use the same URI as the normal get, but add the optional parameters using `@RequestParam(required = false) String author`, that way is more "RESTful" 😄

Reply:
ok, understood, thanks